### PR TITLE
Gen 1261 update claimer to dynamically compute decay constant from

### PR DESCRIPTION
**Pull request by asselstine**

### CODE REVIEW

File: src/Claimer.sol [hunk 1]
```diff
@@ -277,12 +255,35 @@ contract Claimer {
...
+  /// @return The starting fee for prize claims
+  function _computeFeeTarget() internal view returns (uint256) {
+    uint8 numberOfTiers = prizePool.numberOfTiers();
+    // we expect the fee to match the second canary tier
+    return prizePool.getTierPrizeSize(numberOfTiers - 2);
+  }
+
+  /// @notice Computes the decay constant for the VRGDA.
+  /// @dev This is a decay constant that ensures the fee will grow to the target max fee
+  /// @param _targetFee The starting fee
```
Comment (trmid) [inline src/Claimer.sol]:
```suggestion
  /// @param _targetFee The target fee
```

File: src/Claimer.sol [hunk 2]
```diff
@@ -277,12 +255,35 @@ contract Claimer {
...
     if (prizePool.isCanaryTier(_tier)) {
-      return prizePool.getTierPrizeSize(_tier);
+      return type(uint256).max; // no limit
     } else {
       return _computeMaxFee(_tier);
     }
   }
 
+  /// @notice Compute the starting fee for prize claims
+  /// @return The starting fee for prize claims
```
Comment (trmid) [inline src/Claimer.sol]:
```suggestion
  /// @return The target fee for prize claims
```

File: src/Claimer.sol [hunk 3]
```diff
@@ -295,23 +296,23 @@ contract Claimer {
   }
 
   /// @notice Computes the fee for the next claim.
-  /// @param _minimumFee The minimum fee that should be charged
+  /// @param _targetFee The minimum fee that should be charged
```
Comment (trmid) [inline src/Claimer.sol]:
```suggestion
  /// @param _targetFee The target fee that should be charged
```

File: src/Claimer.sol [hunk 4]
```diff
@@ -277,12 +255,35 @@ contract Claimer {
...
+    // we expect the fee to match the second canary tier
+    return prizePool.getTierPrizeSize(numberOfTiers - 2);
+  }
+
+  /// @notice Computes the decay constant for the VRGDA.
+  /// @dev This is a decay constant that ensures the fee will grow to the target max fee
+  /// @param _targetFee The starting fee 
+  /// @return The decay constant
+  function _computeDecayConstant(uint256 _targetFee) internal view returns (SD59x18) {
+    uint maximumFee = _computeMaxFee(0);
```
Comment (trmid) [inline src/Claimer.sol:267]: Would it help the auction price discovery if we multiplied this by the `maxFeePortionOfPrize` fraction?

File: src/Claimer.sol [hunk 5]
```diff
@@ -277,12 +255,35 @@ contract Claimer {
...
+  /// @notice Compute the starting fee for prize claims
+  /// @return The starting fee for prize claims
+  function _computeFeeTarget() internal view returns (uint256) {
+    uint8 numberOfTiers = prizePool.numberOfTiers();
+    // we expect the fee to match the second canary tier
+    return prizePool.getTierPrizeSize(numberOfTiers - 2);
+  }
+
+  /// @notice Computes the decay constant for the VRGDA.
+  /// @dev This is a decay constant that ensures the fee will grow to the target max fee
```
Comment (trmid) [inline src/Claimer.sol]: This is a bit confusing. Is it growing to the target fee or the max fee?

File: src/Claimer.sol [hunk 6]
```diff
@@ -277,12 +255,35 @@ contract Claimer {
...
   function computeMaxFee(uint8 _tier) public view returns (uint256) {
     if (prizePool.isCanaryTier(_tier)) {
-      return prizePool.getTierPrizeSize(_tier);
+      return type(uint256).max; // no limit
     } else {
       return _computeMaxFee(_tier);
     }
   }
 
+  /// @notice Compute the starting fee for prize claims
```
Comment (trmid) [inline src/Claimer.sol]:
```suggestion
  /// @notice Compute the target fee for prize claims
```

**Review by trmid — CHANGES_REQUESTED** [review]

File: src/Claimer.sol (hunk 4 above)
Comment (asselstine) [reply src/Claimer.sol:267]: Ah yes!  It definitely should; that would make it more efficient

Comment (asselstine) [reply src/Claimer.sol:267]: Ah wait, it already is in the `_computeMaxFee` function

**Pull request merged**